<?xml version="1.0"?>
<!--
config_nodepth.launch
Author: Ravi Joshi
Note: Do not run this file directly. Please call run.launch file instead.
Date: 2019/11/23
-->
<launch>

    <!-- usb_acmノードを実行する -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
        <!-- カメラの指定　defaultは/dev/video0 -->
        <param name="video_device" value="/dev/video0"/>
        <!-- 画像の大きさ（横幅）の指定 -->
        <param name="image_width" value="640"/>
        <!-- 画像の大きさ（縦幅）の指定 -->
        <param name="image_height" value="480"/>
        <!-- 画像フォーマット設定　"mjpeg" or "yuyv" or "uyvy" -->
        <param name="pixel_format" value="mjpeg"/>
        <!-- fpsの指定　frame per second -->
        <param name="framerate" value="30"/>
        <!-- コントラストの設定（色設１） -->
        <param name="contrast" value="32"/>
        <!-- 明るさの設定（色設２） -->
        <param name="brightness" value="32"/>
        <!-- 彩度の設定（色設３） -->
        <param name="saturation" value="32"/>
        <!-- 自動フォーカス機能の設定　ブール値 -->
        <param name="autofocus" value="true"/>
        <!-- フォーカス位置の設定 -->
        <param name="focus" value="51"/>
    </node>

  <!-- rostopic to subscribe for color images -->
  <arg name="color_topic" default="/usb_cam/image_raw"/>

  <!-- rostopic to subscribe for depth images.
  leave this field as it is because we do not have depth information. -->
  <arg name="depth_topic" default="depth_topic" />

  <!-- rostopic to subscribe for camera calibration parameters.
  leave this field as it is because we do not have depth information. -->
  <arg name="cam_info_topic" default="cam_info_topic" />

  <!-- frame id for point cloud and 3D skeleton data.
  leave this field as it is because we do not have depth information. -->
  <arg name="frame_id" default="no_depth" />

  <!-- we do not have depth information.
  hence set the flag. -->
  <arg name="no_depth" default="true" />
  <arg name="skeleton" default="false" />
  <arg name="skeleton_hands" default="false" />

  <!-- no RViz support -->
  <arg name="rviz" default="true" />
  

  <!-- note that the following parameters are going to assigned by other launch file upon call. -->
  <arg name="print" default="" />
  <arg name="pub_topic" default="" />
  <arg name="pointcloud" default="" />
  <arg name="id_text_size" default="" />
  <arg name="openpose_args" default="" />
  <arg name="id_text_offset" default="" />
  <arg name="skeleton_line_width" default="" />
  <arg name="synchronous" default="" />
  <arg name="py_openpose_path" default="" />


  <include file="$(find ros_openpose)/launch/core.launch" >
    <arg name="print" value="$(arg print)" />
    <arg name="no_depth" value="$(arg no_depth)" />
    <arg name="skeleton" value="$(arg skeleton)" />
    <arg name="frame_id" value="$(arg frame_id)" />
    <arg name="pub_topic" value="$(arg pub_topic)" />
    <arg name="color_topic" value="$(arg color_topic)" />
    <arg name="depth_topic" value="$(arg depth_topic)" />
    <arg name="id_text_size" value="$(arg id_text_size)" />
    <arg name="openpose_args" value="$(arg openpose_args)" />
    <arg name="id_text_offset" value="$(arg id_text_offset)" />
    <arg name="skeleton_hands" value="$(arg skeleton_hands)" />
    <arg name="cam_info_topic" value="$(arg cam_info_topic)" />
    <arg name="skeleton_line_width" value="$(arg skeleton_line_width)" />
    <arg name="synchronous" value="$(arg synchronous)" />
    <arg name="py_openpose_path" value="$(arg py_openpose_path)" />
  </include>
</launch>
